peps/pep-0741.rst

```diff
@@ -0,0 +1,541 @@
+(August 2022).
+
+At the end, it was decided to not add a new ``PyConfig`` member in
+stable branches, but only add a new ``PyConfig.int_max_str_digits``
+member to the development branch (which became Python 3.12). A dedicated
+private global variable (unrelated to ``PyConfig``) is used in stable
+branches.
+
+Redundancy between PyPreConfig and PyConfig
+-------------------------------------------
+
+The Python preinitialization uses the ``PyPreConfig``  structure and the
+Python initialization uses the ``PyConfig`` structure. Both structures
+have 4 duplicated members: ``dev_mode``, ``parse_argv``, ``isolated``
+and ``use_environment``.
+
+The redundancy is caused by the fact that the two structures are
+separated, whereas some ``PyConfig`` members are needed by the
+preinitialization.
+
+
+Specification
+=============
+
+C API:
+
+* ``PyInitConfig`` structure
+* ``PyInitConfig_Python_New()``
+* ``PyInitConfig_Isolated_New()``
+* ``PyInitConfig_Free(config)``
+* ``PyInitConfig_SetInt(config, name, value)``
+* ``PyInitConfig_SetStr(config, name, value)``
+* ``PyInitConfig_SetWStr(config, name, value)``
+* ``PyInitConfig_SetStrList(config, name, length, items)``
+* ``PyInitConfig_SetWStrList(config, name, length, items)``
+* ``Py_InitializeFromInitConfig(config)``
+* ``PyInitConfig_Exception(config)``
+* ``PyInitConfig_GetError(config, &err_msg)``
+* ``PyInitConfig_GetExitCode(config, &exitcode)``
+* ``Py_ExitWithInitConfig(config)``
+* ``PyConfig_Get(name)``
+* ``PyConfig_GetInt(name, &value)``
+
+Python API:
+
+* ``sys.get_config(name)``
+
+The C API uses null-terminated UTF-8 encoded strings to refer to a
+configuration option.
+
+All C API functions are added to the limited C API version 3.13.
+
+The ``PyInitConfig`` structure is implemented by combining the 3
+structures of the ``PyConfig`` API:
+
+* ``PyPreConfig preconfig``
+* ``PyConfig config``
+* ``PyStatus status``
+
+The ``PyStatus`` status is no longer separated, but part of the unified
+``PyInitConfig`` structure, which makes the API easier to use.
+
+
+Configuration Options
+---------------------
+
+Configuration options names are names of ``PyPreConfig`` and
+``PyConfig`` structure members such as ``"verbose"``
+(``PyConfig.verbose``), ``"buffered_stdio"``
+(``PyConfig.buffered_stdio``), or ``"allocator"``
+(``PyPreConfig.allocator``).
+
+The type of configuration options depends on the option. For example,
+``"verbose"`` option type is an integer, whereas
+``"module_search_paths"`` option type is an array of wide strings.
+
+See the `PyConfig documentation
+<https://docs.python.org/dev/c-api/init_config.html#pyconfig>`_
+and the `PyPreConfig documentation
+<https://docs.python.org/dev/c-api/init_config.html#pypreconfig>`_.
+
+
+Configure the Python initialization
+-----------------------------------
+
+``PyInitConfig`` structure:
+
+    Opaque structure to configure the Python preinitialization and the
+    Python initialization.
```

Comment:
Do you want these as italicised blockquotes?

![image](https://github.com/python/peps/assets/1324225/5bd478bb-aab3-4a13-a8cb-dac30e21ffb4)

If not, deleting the first empty line:

```suggestion
``PyInitConfig`` structure:
    Opaque structure to configure the Python preinitialization and the
    Python initialization.
```

Gives a "definition list":

![image](https://github.com/python/peps/assets/1324225/9a08d978-399d-481a-ab07-42440cb7efdc)

Hmm, maybe the original gives a bit more space to read.